akka-stream-tests/src/test/scala/akka/stream/scaladsl/QueueSinkSpec.scala

```diff
@@ -112,5 +114,26 @@ class QueueSinkSpec extends AkkaSpec {
       queue.pull().onFailure { case e ⇒ e.isInstanceOf[IllegalStateException] should ===(true) }
     }
 
+    "keep on sending even after the buffer has been full" in assertAllStagesStopped {
+      val (probe, queue) = Source(1 to 20)
+        .alsoToMat(Flow[Int].take(16).watchTermination()(Keep.right).to(Sink.ignore))(Keep.right)
+        .toMat(Sink.queue())(Keep.both)
+        .run()
+
+      // wait for the buffer to fill up before pulling
+      akka.pattern.after(0.5 seconds, system.scheduler)(queue.pull()) pipeTo testActor
```

Comment:
@johanandren this would pull immediately, which is not what we want: we need to wait until the buffer has filled up. What is missing here is to wait for the `probe` first, and the proceed.